File: Leetcode_205.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        u = list(s)
        v = list(t)

        pattern_s = []
        pattern_t = []

        for i in range(len(u)):
            pattern_s.append(u.index(u[i]))
            pattern_t.append(v.index(v[i]))

        return pattern_s == pattern_t

File: Leetcode_205.java
class Solution {
    public boolean isIsomorphic(String s, String t) {
        List<Integer> patternS = new ArrayList<>();
        List<Integer> patternT = new ArrayList<>();

        for (int i = 0; i < s.length(); i++) {
            patternS.add(s.indexOf(s.charAt(i)));
            patternT.add(t.indexOf(t.charAt(i)));
        }

        return patternS.equals(patternT);
    }
